enhancements/machine-config/enable-evented-pleg-in-openshift.md

```diff
@@ -0,0 +1,196 @@
+
+## Motivation
+
+Evented pleg requires enabling of the flags present in both the Kubelet and the cri-o command line arguments.
+So, one of the motivations of this proposal is to make this process easier. User can update both the above configurations by just updating a single field of a custome resource object. 
+Other main motive is that the Evented PLEG feature can play an important role in the Kubelet's performance (compared to the existing Generic PLEG) in the presence of huge number of pods on a given node.
+Hence, this feature pushes the Openshift's ability to achieve greater pod density on a node. 
+
+
+### User Stories
+
+> "As an OCP user, I want to enable the evented pleg feature in the Openshift clusters so that the Kubelet's performance while determining the pod life-cycle improves"
+
+### Goals
+- Support/Enable evented pleg feature in the Openshift 
+### Non-Goals
+
+- Eliminate the Generic PLEG completely and solely rely on the Evented PLEG
+
+## Proposal
+
+An option to enable the evented pleg will have to reside in a node specific configuration object as the enabling requires the configuration changes in the Kubelet, CRI-O configs.
+[OpenShift Node config](https://github.com/openshift/api/blob/master/config/v1/types_node.go#L19) (nodes.config.openshift.io) object contains some of the node related configurations such as CgroupMode, WorkerlatencyProfiles.
+This nodes.config object can be a good fit to have an extra field to rely on enabling the evented pleg feature.
+This object is monitored by the MCO in order to make the change in the kubelet as well as cri-o configuration on all the nodes of the cluster.
+
+### Workflow Description
+- Add an extra field to enable evented pleg in the nodes.config API object. 
+- Monitor the above field via Kubelet Config controller and the Container Runtime controller present in the MCO
+- Machine config objects get created with the updated kubelet and the cri-o configs.
+- Openshift Cluster nodes get rebooted and updated with the evented pleg feature enabled. 
+
+### API Extensions
+
+Create an additional field within the nodes.config object named `EnablePodEvents` :
+
+```go
+package v1
+
+type Node struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// spec holds user settable values for configuration
+	// +kubebuilder:validation:Required
+	// +required
+	Spec NodeSpec `json:"spec"`
+
+	// status holds observed values.
+	// +optional
+	Status NodeStatus `json:"status"`
+}
+
+
+type NodeSpec struct {
+	// CgroupMode
+
+	// WorkerLatencyProfile
+
+	// EnablePodEvents determines whether to enable the evented pleg feature to
+	// send the container events from the CRI-O through gRPC to the Kubelet
+	// +optional
+	EnablePodEvents bool `json:"enablePodEvents,omitempty"`
```

Comment:
Naming this EventedPleg seems ok since this is a user facing structure. Thoughts?

Reply:
Yes, I agree, Updated.
Just took a naming reference from the cri-o config to enable evented pleg.